src/session/session_dhandle.c

```diff
@@ -570,18 +569,18 @@ __wt_session_get_btree_ckpt(WT_SESSION_IMPL *session, const char *uri, const cha
      * Applications can use the internal reserved name "WiredTigerCheckpoint" to open the latest
      * checkpoint, but they are not allowed to directly open specific checkpoint versions, such as
      * "WiredTigerCheckpoint.6".
+     *
+     * There is no such restriction for the verification flow.
      */
     is_reserved_name = cval.len > strlen(WT_CHECKPOINT) && WT_PREFIX_MATCH(cval.str, WT_CHECKPOINT);
-    if (is_reserved_name)
+    if (is_reserved_name && !(flags & WT_BTREE_VERIFY))
         WT_RET_MSG(
           session, EINVAL, "the prefix \"%s\" for checkpoint cursors is reserved", WT_CHECKPOINT);
 
     /*
-     * Test for the internal checkpoint name (WiredTigerCheckpoint). Note: must_resolve is true in a
-     * subset of the cases where is_unnamed_ckpt is true.
+     * Test for the internal checkpoint name (WiredTigerCheckpoint).
      */
     must_resolve = WT_CONFIG_LIT_MATCH(WT_CHECKPOINT, cval);
-    is_unnamed_ckpt = cval.len >= strlen(WT_CHECKPOINT) && WT_PREFIX_MATCH(cval.str, WT_CHECKPOINT);
```

Comment:
AFAIS `must_resolve` and `is_unnamed_ckpt` are always the same since we filter all the specific WT checkpoints name before by checking `is_reserved_name`. Please double check it and let me know if I am wrong :)